src/music.cpp

```diff
@@ -0,0 +1,86 @@
+#include "music.h"
+#include <fstream>
+#include <iostream>
+
+Asound::Asound() {
+  std::ifstream file(MUSIC_CONFIG_PATH + "config.json");
+
+  if (!file.is_open()) {
+    std::cerr << "Failed to open config file for music" << std::endl;
+    exit(1);
+  }
+
+  file >> config;
+
+  for (auto &[k, v] : config.items()){
+    std::cout << "Read sound key: " << k << " val: " << v << std::endl;
+    std::string s = MUSIC_CONFIG_PATH + std::string(v);
+    sounds.emplace(k, s);
+  }
+}
+
+Asound::~Asound() {}
+
+void Asound::play(std::string sound){
+  sounds[sound].Play();
```

Comment:
In C++, the index operator on a map constructs the object if it is not found. This does not work, because there is no zero arugment constructor for `raylib::Sound`.

to fix this I offer a few solutions.

```C++
// this will segfault if not found
sounds.find(sound)->second.Play()
```

```C++
raylib::Sound& Asound::operator[](const std::string &sound) {
    if (sounds.find(sound) == sounds.end())
        throw std::runtime_error("sound not found");

    return sounds.find(sound)->second
}
```

Reply:
thanks, this worked :100: